### PR TITLE
enable DPDK meson-ninja build

### PR DESCRIPTION
**Pull request by vipinpv85**
Following patches brings in code changes to enable dpdk meson-ninja to build libraries to DPVS. This allows an easy migration path to using meson-ninja as DPDK has deprecated the support for Makefile from 20.11 onwards. Fallback mechanism allows check for RTE_SDK in `dpdk.mk` to ensure the compatibility with older releases too.

1. DPDK version: 18.11 LTS
2. gcc: Ubuntu 7.5.0-3ubuntu1~18.04

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,7 +91,18 @@ $ ...
...
+1. Using meson-ninja for building DPDK libraries. The `dpdk.mk` checks for presence of libdpdk
+
+```bash
+$ cd dpdk-stable-18.11.11
+$ mkdir [user desired install folder - dpdklib]
+$ mkdir [user desired build folder - dpdkbuild]
+$ meson  -Dprefix=[dpdklib] dpdkbuild
+$ ninja -C dpdkbuild
+$ cd dpdkbuild; ninja install
+```
```
Comment (ywc689) [inline README.md:103]: I think it's helpful to add `libdpdk.pc` to pkg-config search path.

```
...
$ cd dpdkbuild; ninja install
$ export PKG_CONFIG_PATH=${PKG_CONFIG_PATH:+$PKG_CONFIG_PATH:}$(pwd)/../dpdklib/lib64/pkgconfig/libdpdk.pc
```

File: src/Makefile [hunk 2]
```diff
@@ -73,7 +76,7 @@ all: $(TARGET)
 
 $(TARGET): $(OBJS)
 	@echo "  $(notdir $@)"
-	$(Q)$(CC) $(CFLAGS) $^ -o $@
+	$(Q)$(CC) $(CFLAGS) $^ $(LIBS) -o $@
```
Comment (ywc689) [inline src/Makefile:79]: ` $(LIBS)` should be removed from `CFLAGS` in line 70. It's the reason why the CI complained a failure.

**Review by ywc689 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vipinpv85) [reply README.md:103]: agreed,

File: src/Makefile (hunk 2 above)
Comment (vipinpv85) [reply src/Makefile:79]: @ywc689 I also agree with you, but this is not introduced by current changes. This looks like a regression issue from previous commits. 

But can change for the community

**Review by vipinpv85 — COMMENTED** [review]
updated the PKG_CONFIG_PATH in readme and made LIBS generic to Makefile and Meson-NINJA

**Review by vipinpv85 — COMMENTED** [review]
addressed comments

**Comment by ywc689** [comment]
meson/ninja build is supported by pr #737 based on this patch now.

**Pull request closed without merging**